### ISSUE TITLE
Rule Change: no-empty suggest to add comments inside empty blocks 

### ISSUE THREAD

**Issue by amareshsm** [eslint/eslint#15899]
### What rule do you want to change?

no-empty

### What change to do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
const foo = () => {
  console.log('Still going');
  return Math.random() > 0.5;
};

while(foo()) {}
```

### What does the rule currently do for this code?

Throws an error because an empty while loop is not allowed.

### What will the rule do after it's changed?

show a suggestion to add comments inside an empty loop. 

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Reference [#15839 ](https://github.com/eslint/eslint/issues/15839#issuecomment-1119300679)

**Comment by mdjermanovic** [comment]
Thanks for the proposal!

> show a suggestion to add comments inside an empty while loop.

I have two questions:

1. This rule disallows empty block statements. Is the proposal to provide suggestions for all empty block statements, or only for those in while loops?
2. How will the suggestion fix the code?

**Comment by mdjermanovic** [comment]
> 1. This rule disallows empty block statements. Is the proposal to provide suggestions for all empty block statements, or only for those in while loops? >>  In while loops alone.

Why only while loops?

**Comment by amareshsm** [comment]
I think we can show suggestion for all empty block statements.

**Comment by mdjermanovic** [comment]
Since this rule allows block statements with a comment inside regardless of the context in which the block statement appears (as in [this demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLWVtcHR5OiBcImVycm9yXCIgKi9cblxue1xuICAvKiBlbXB0eSAqL1xufVxuXG5pZiAoZm9vKSB7XG4gIC8qIGVtcHR5ICovXG59IGVsc2Uge1xuICAvKiBlbXB0eSAqL1xufVxuXG53aGlsZSAoZm9vKSB7XG4gIC8qIGVtcHR5ICovXG59Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==)), I think it would make the most sense to always provide this suggestion, without checking whether the block statement is a loop body.

**Comment by amareshsm** [comment]
I agree we can suggest adding comments to all empty block statements.

**Comment by mdjermanovic** [comment]
Suggestion that adds `/* empty */` comment inside the reported empty block statement makes sense to me :+1: 

Since all new features need a consensus from the team, let's wait for more opinions.

**Comment by nzakas** [comment]
Makes sense to me.

**Comment by amareshsm** [comment]
I am working on this. I will send a PR soon.
